states/remote/remote_test.go

```diff
@@ -126,3 +116,54 @@ func (c *mockClient) appendLog(method string, content []byte) {
 	}
 	c.log = append(c.log, mockClientRequest{method, contentVal})
 }
+
+// mockRemoteClient is like mockClient, but also implements
+// EnableForcePush, allowing testing for this behavior
+type mockRemoteClient struct {
+	current []byte
+	force   bool
+	log     []mockClientRequest
+}
+
+func (c *mockRemoteClient) Get() (*Payload, error) {
```

Comment:
Not all these methods are used in the tests, so are extraneous for mockRemoteClient, however I've included them for completeness if/when this client might be re-used for tests covering other differentiation with the remote backend.

Reply:
Since these are both (nominally) mock remote clients, what do you think about naming your new one something like `mockRemoteClientForcePusher` or `mockClientForcePusher` (`ClientForcePusher` being the interface it implements)? 

I genuinely don't know if this would make it more or less confusing, so absolutely feel free to disregard this.